nixos/modules/services/desktops/espanso.nix

```diff
@@ -0,0 +1,40 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.services.espanso;
+in {
+  options = {
+    services.espanso = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable espanso
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.espanso;
+        description = ''
+          Espanso package to use
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.espanso = {
+      enable = true;
+      description = "Espanso daemon";
+      path = with pkgs; [ cfg.package libnotify xclip ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/espanso daemon";
+        Restart = "on-failure";
+      };
+      wantedBy = [ "default.target" ];
+    };
+
+    environment.systemPackages = [ cfg.package ];
```

Comment:
Does `espanso` need to be in `PATH` if you're running a `systemd` user service?

Reply:
It does, to be able to talk to it for some functions.

https://espanso.org/docs/configuration/